shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.m

```diff
@@ -23,4 +41,200 @@ - (void)testBinaryMessenger {
+                               return [message[@"platformContrast"] isEqualToString:@"normal"];
+                             }]]);
+
+  // Clean up mocks
+  [engine stopMocking];
+  [settingsChannel stopMocking];
+}
+
+- (void)testItReportsPlatformContrastWhenViewWillAppear {
+  if (!@available(iOS 13, *)) {
+    return;
+  }
+
+  // Setup test.
+  id engine = OCMClassMock([FlutterEngine class]);
+
+  id settingsChannel = OCMClassMock([FlutterBasicMessageChannel class]);
+  OCMStub([engine settingsChannel]).andReturn(settingsChannel);
+
+  FlutterViewController* vc = [[FlutterViewController alloc] initWithEngine:engine
+                                                                    nibName:nil
+                                                                     bundle:nil];
+
+  // Exercise behavior under test.
+  [vc viewWillAppear:false];
+
+  // Verify behavior.
+  OCMVerify([settingsChannel sendMessage:[OCMArg checkWithBlock:^BOOL(id message) {
+                               return [message[@"platformContrast"] isEqualToString:@"normal"];
+                             }]]);
+
+  // Clean up mocks
+  [engine stopMocking];
+  [settingsChannel stopMocking];
+}
+
+- (void)testItReportsHighContrastWhenTraitCollectionRequestsIt {
+  if (!@available(iOS 13, *)) {
+    return;
+  }
+
+  // Setup test.
+  id engine = OCMClassMock([FlutterEngine class]);
+
+  id settingsChannel = OCMClassMock([FlutterBasicMessageChannel class]);
+  OCMStub([engine settingsChannel]).andReturn(settingsChannel);
+
+  FlutterViewController* realVC = [[FlutterViewController alloc] initWithEngine:engine
+                                                                        nibName:nil
+                                                                         bundle:nil];
+  id mockTraitCollection = [self setupFakeTraitCollectionWithContrast:UIAccessibilityContrastHigh];
+
+  // We partially mock the real FlutterViewController to act as the OS and report
+  // the UITraitCollection of our choice. Mocking the object under test is not
+  // desirable, but given that the OS does not offer a DI approach to providing
+  // our own UITraitCollection, this seems to be the least bad option.
+  id partialMockVC = OCMPartialMock(realVC);
+  OCMStub([partialMockVC traitCollection]).andReturn(mockTraitCollection);
+
+  // Exercise behavior under test.
+  [partialMockVC traitCollectionDidChange:mockTraitCollection];
+
+  // Verify behavior.
+  OCMVerify([settingsChannel sendMessage:[OCMArg checkWithBlock:^BOOL(id message) {
+                               return [message[@"platformContrast"] isEqualToString:@"high"];
+                             }]]);
+
+  // Clean up mocks
+  [partialMockVC stopMocking];
+  [engine stopMocking];
+  [settingsChannel stopMocking];
+  [mockTraitCollection stopMocking];
+}
+
+- (UITraitCollection*)setupFakeTraitCollectionWithContrast:(UIAccessibilityContrast)contrast {
```

Comment:
s/setupFakeTrait.../fakeTrait.../

This isn't operating on a trait collection, it's returning one, so should be named like a getter.

Also, this should get a brief declaration comment as well.

Reply:
done